tuf/ngclient/updater.py

```diff
@@ -401,17 +401,19 @@ def _preorder_depth_first_walk(
         # List of delegations to be interrogated. A (role, parent role) pair
         # is needed to load and verify the delegated targets metadata.
         delegations_to_visit = [("targets", "root")]
-        visited_role_names: Set[Tuple[str, str]] = set()
+        visited_role_names: Set[str] = set()
         number_of_delegations = self.config.max_delegations
 
         # Preorder depth-first traversal of the graph of target delegations.
         while number_of_delegations > 0 and len(delegations_to_visit) > 0:
 
             # Pop the role name from the top of the stack.
             role_name, parent_role = delegations_to_visit.pop(-1)
+            if parent_role not in visited_role_names:
+                visited_role_names.add(parent_role)
```

Comment:
I don't think this is needed. We always start from "targets" and do a pre-order depth-first search, if the traversal code is correct, the parent is always visited before the child.